### ISSUE TITLE
Error logging in D3D12 functions

### ISSUE THREAD

**Issue by Nielsbishere** [TeamWisp/WispRenderer#353]
**Is your request related to a problem? Please describe.**
Most d3d12 functions don't do error checking. 

**Describe the solution you'd like**
For example mapping a buffer should use a function similar to Microsoft example code (ThrowIfError). Turning a HRESULT and outputting it to log.

**Describe alternatives you've considered**
**Additional context**
N.A.

**Comment by VZout** [comment]
We do do error checking. `TRY` and `TRY_M`.

**Comment by Nielsbishere** [comment]
Yeah but not everywhere, an example is in mapping/unmapping memory. If this fails, it memcpys into a nullptr. Might be good to have it for debugging, because mapping memory fails if there's no shared memory left.

**Comment by VZout** [comment]
in the case of mapping and unmapping it shouldn't do error handling. the caller should do the error handling. that way you don't need to do as many checks,.

**Comment by Nielsbishere** [comment]
I'd rather have 1 compare and added safety than saving that and having to do it in the caller
